{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"leftoverDataWhenDone"},{"kind":"text","text":"("},{"kind":"typeIdentifier","identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/ByteBuffer","preciseIdentifier":"s:3NIO10ByteBufferV","text":"ByteBuffer"},{"kind":"text","text":")"}],"languages":["swift"],"platforms":["macOS"]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/nio\/bytetomessagedecodererror\/leftoverdatawhendone(_:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/org.swift.docc\/documentation\/NIO\/ByteToMessageDecoderError\/leftoverDataWhenDone(_:)","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"This error can be thrown by "},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":"s if there was unexpectedly some left-over data when the"},{"type":"text","text":" "},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":" was removed from the pipeline or the "},{"type":"codeVoice","code":"Channel"},{"type":"text","text":" was closed."}],"kind":"symbol","metadata":{"fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"leftoverDataWhenDone"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"ByteBuffer","preciseIdentifier":"s:3NIO10ByteBufferV"},{"kind":"text","text":")"}],"title":"ByteToMessageDecoderError.leftoverDataWhenDone(_:)","roleHeading":"Case","role":"symbol","symbolKind":"case","externalID":"s:3NIO25ByteToMessageDecoderErrorO20leftoverDataWhenDoneyAcA0B6BufferVcACmF","modules":[{"name":"NIO"}]},"hierarchy":{"paths":[["doc:\/\/org.swift.docc\/documentation\/NIO","doc:\/\/org.swift.docc\/documentation\/NIO\/ByteToMessageDecoderError"]]},"references":{"doc://org.swift.docc/documentation/NIO/ByteToMessageDecoderError":{"role":"symbol","title":"ByteToMessageDecoderError","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"ByteToMessageDecoderError"}],"abstract":[{"type":"text","text":"Common errors thrown by "},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":"s."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/ByteToMessageDecoderError","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"ByteToMessageDecoderError"}],"url":"\/documentation\/nio\/bytetomessagedecodererror"},"doc://org.swift.docc/documentation/NIO/ByteBuffer":{"role":"symbol","title":"ByteBuffer","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"ByteBuffer"}],"abstract":[{"type":"codeVoice","code":"ByteBuffer"},{"type":"text","text":" stores contiguously allocated raw bytes. It is a random and sequential accessible sequence of zero or"},{"type":"text","text":" "},{"type":"text","text":"more bytes (octets)."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/ByteBuffer","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"ByteBuffer"}],"url":"\/documentation\/nio\/bytebuffer"},"doc://org.swift.docc/documentation/NIO":{"role":"collection","title":"NIO","abstract":[],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO","kind":"symbol","type":"topic","url":"\/documentation\/nio"},"doc://org.swift.docc/documentation/NIO/ByteToMessageDecoderError/leftoverDataWhenDone(_:)":{"role":"symbol","title":"ByteToMessageDecoderError.leftoverDataWhenDone(_:)","fragments":[{"kind":"keyword","text":"case"},{"kind":"text","text":" "},{"kind":"identifier","text":"leftoverDataWhenDone"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"ByteBuffer","preciseIdentifier":"s:3NIO10ByteBufferV"},{"kind":"text","text":")"}],"abstract":[{"type":"text","text":"This error can be thrown by "},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":"s if there was unexpectedly some left-over data when the"},{"type":"text","text":" "},{"type":"codeVoice","code":"ByteToMessageDecoder"},{"type":"text","text":" was removed from the pipeline or the "},{"type":"codeVoice","code":"Channel"},{"type":"text","text":" was closed."}],"identifier":"doc:\/\/org.swift.docc\/documentation\/NIO\/ByteToMessageDecoderError\/leftoverDataWhenDone(_:)","kind":"symbol","type":"topic","url":"\/documentation\/nio\/bytetomessagedecodererror\/leftoverdatawhendone(_:)"}}}